packages/web3-eth-contract/src/contract.ts

```diff
@@ -955,37 +972,73 @@ export class Contract<Abi extends ContractAbi>
 		this._jsonInterface = [...result] as unknown as ContractAbiWithSignature;
 	}
 
+	// eslint-disable-next-line class-methods-use-this
+	private _getAbiParams(abi: AbiFunctionFragment, params: unknown[]): Array<unknown> {
+		try {
+			return validatorUtils.transformJsonDataToAbiFormat(abi.inputs ?? [], params);
+		} catch (error) {
+			throw new Web3ContractError(
+				`Invalid parameters for method ${abi.name}: ${(error as Error).message}`,
+			);
+		}
+	}
 	private _createContractMethod<T extends AbiFunctionFragment, E extends AbiErrorFragment>(
 		abi: T,
 		errorsAbis: E[],
 	): ContractBoundMethod<T> {
 		return (...params: unknown[]) => {
 			let abiParams!: Array<unknown>;
+			const abis = this._overloadedMethodAbis.get(abi.name) ?? [];
+			let methodAbi: AbiFunctionFragment = abis[0];
 
-			try {
-				abiParams = validatorUtils.transformJsonDataToAbiFormat(abi.inputs ?? [], params);
-			} catch (error) {
-				throw new Web3ContractError(
-					`Invalid parameters for method ${abi.name}: ${(error as Error).message}`,
-				);
-			}
+			const arrayOfAbis: AbiFunctionFragment[] = abis.filter(
+				_abi => (_abi.inputs ?? []).length === params.length,
+			);
 
-			validator.validate(abi.inputs ?? [], abiParams);
+			if (abis.length === 1 || arrayOfAbis.length === 0) {
+				abiParams = this._getAbiParams(methodAbi, params);
+				validator.validate(abi.inputs ?? [], abiParams);
+			} else {
+				const errors: Web3ValidationErrorObject[] = [];
+
+				for (const _abi of arrayOfAbis) {
+					try {
+						abiParams = this._getAbiParams(_abi, params);
+						validator.validate(
+							_abi.inputs as unknown as ValidationSchemaInput,
+							abiParams,
+						);
+						methodAbi = _abi;
+						break;
+					} catch (e) {
+						errors.push(e as Web3ValidationErrorObject);
+					}
+				}
+				if (errors.length === arrayOfAbis.length) {
+					throw new Web3ValidatorError(errors);
+				}
+			}
 
-			if (abi.stateMutability === 'payable' || abi.stateMutability === 'pure') {
+			if (methodAbi.stateMutability === 'payable' || methodAbi.stateMutability === 'pure') {
```

Comment:
This is out of the scope of this MR. But I found it good to ask here.
Why `pure` is handled the same as `payable`? The code for `payable` bellow uses `PayableCallOptions` and `PayableMethodObject` that does not seems to fit with `pure`. Do you think we should just have here:
```suggestion
			if (methodAbi.stateMutability === 'payable') {
```
Or what do you think?

Reply:
in my opinion we can remove this `IF` statement at all

Reply:
yep. you are right we do not need pure in this `if` statement